### PR TITLE
Dyz/crfg

### PR DESCRIPTION
**Pull request by christian-yi**

### CODE REVIEW

**Review by christian-yi — COMMENTED** [review]
recommended to delete useless comments in runtime module

File: runtime/Cargo.toml [hunk 1]
```diff
@@ -27,10 +27,15 @@ runtime-primitives = { package = "sr-primitives", git = "https://github.com/yeec
 client = { package = "substrate-client", git = "https://github.com/yeeco/substrate.git", default_features = false, branch = "yee-v1.0" }
 offchain-primitives = { package = "substrate-offchain-primitives", git = "https://github.com/yeeco/substrate.git", default-features = false, branch = "yee-v1.0" }
 consensus_authorities = { package = "substrate-consensus-authorities", git = "https://github.com/yeeco/substrate.git", default-features = false, branch = "yee-v1.0" }
+crfg = { package = "yee-srml-crfg", path = "../srml/yee-crfg", default-features = false }
+finality-tracker = { package = "srml-finality-tracker", git = "https://github.com/yeeco/substrate.git", default_features = false, branch = "yee-v1.0" }
+#ession = { package = "srml-session", git = "https://github.com/yeeco/substrate.git", default_features = false, branch = "yee-v1.0" }
```
Comment (guoxbin) [inline runtime/Cargo.toml]: remove unused

File: runtime/Cargo.toml [hunk 2]
```diff
@@ -27,10 +27,15 @@ runtime-primitives = { package = "sr-primitives", git = "https://github.com/yeec
...
+#staking = { package = "srml-staking", git = "https://github.com/yeeco/substrate.git", default_features = false, branch = "yee-v1.0" }
```
Comment (guoxbin) [inline runtime/Cargo.toml]: remove unused

File: runtime/src/lib.rs [hunk 3]
```diff
@@ -13,12 +13,18 @@ use primitives::bytes;
 use primitives::{ed25519, sr25519, OpaqueMetadata};
 use runtime_primitives::{
 	ApplyResult, transaction_validity::TransactionValidity, generic, create_runtime_str,
-	traits::{self, NumberFor, BlakeTwo256, Block as BlockT, StaticLookup, Verify}
+	traits::{self, NumberFor, BlakeTwo256, Block as BlockT, DigestFor, StaticLookup, Verify}
 };
+//use runtime_primitives::traits::{
```
Comment (guoxbin) [inline runtime/src/lib.rs]: remove unused

File: runtime/src/lib.rs [hunk 4]
```diff
@@ -193,6 +199,31 @@ impl balances::Trait for Runtime {
...
+	type SessionKey = AuthorityId;
+	type Log = Log;
+	type Event = Event;
+}
+
+impl finality_tracker::Trait for Runtime {
+	type OnFinalizationStalled = crfg::SyncedAuthorities<Runtime>;
+}
+
+//impl session::Trait for Runtime {
```
Comment (guoxbin) [inline runtime/src/lib.rs]: remove unused

File: src/chain_spec.rs [hunk 5]
```diff
@@ -2,13 +2,16 @@ use {
...
         sr25519, Pair,
         crypto::Ss58Codec,
+		ed25519::Public as AuthorityId,
+		ed25519,
     },
 };
 use yee_runtime::{
 	AccountId, GenesisConfig, ConsensusConfig, TimestampConfig, BalancesConfig,
-	IndicesConfig,
+	IndicesConfig,CrfgConfig,
```
Comment (guoxbin) [inline src/chain_spec.rs]: add blank after comma

File: src/service.rs [hunk 6]
```diff
@@ -65,9 +72,10 @@ pub struct NodeConfig {
     pub mine: bool,
 }
 
-impl Default for NodeConfig {
+impl<F: substrate_service::ServiceFactory> Default for NodeConfig<F> {
```
Comment (guoxbin) [inline src/service.rs:75]: substrate_service::ServiceFactory is imported, no need add "substrate_service::"

File: src/service.rs [hunk 7]
```diff
@@ -144,31 +153,34 @@ construct_simple_protocol! {
...
-            |service: Self::FullService, executor: TaskExecutor, key: Option<Arc<Pair>>| {
-                if let Some(key) = key {
-                    info!("Using authority key {}", key.public());
-                    let proposer = Arc::new(ProposerFactory {
-                        client: service.client(),
-                        transaction_pool: service.transaction_pool(),
-                        inherents_pool: service.inherents_pool(),
-                    });
-                    let client = service.client();
+	struct Factory {
```
Comment (guoxbin) [inline src/service.rs]: use 4 blanks to indent as master

File: Cargo.lock [hunk 8]
```diff
@@ -739,6 +739,19 @@ dependencies = [
  "libc 0.2.58 (registry+https://github.com/rust-lang/crates.io-index)",
 ]
 
+[[package]]
+name = "finality-grandpa"
```
Comment (guoxbin) [inline Cargo.lock:754]: wrong name

File: Cargo.lock [hunk 9]
```diff
@@ -894,6 +907,17 @@ dependencies = [
  "regex 1.1.7 (registry+https://github.com/rust-lang/crates.io-index)",
 ]
 
+[[package]]
+name = "grandpa-primitives"
```
Comment (guoxbin) [inline Cargo.lock]: wrong name

File: Cargo.lock [hunk 10]
```diff
@@ -5235,6 +5333,33 @@ dependencies = [
  "substrate-primitives 1.0.0 (git+https://github.com/yeeco/substrate.git?branch=yee-v1.0)",
 ]
 
+[[package]]
+name = "yee-crfg-grandpa"
```
Comment (guoxbin) [inline Cargo.lock]: wrong name

File: Cargo.lock [hunk 11]
```diff
@@ -4148,6 +4202,49 @@ dependencies = [
  "ws 0.7.9 (registry+https://github.com/rust-lang/crates.io-index)",
 ]
 
+[[package]]
+name = "substrate-test-client"
+version = "1.0.0"
+source = "git+https://github.com/yeeco/substrate.git?branch=yee-v1.0#99e45db062bb28b2ce93e41286d89196c5d74ad9"
```
Comment (guoxbin) [inline Cargo.lock]: all 99e4 are wrong, master is 643d, https://github.com/yeeco/substrate/commits/yee-v1.0

**Review by guoxbin — CHANGES_REQUESTED** [review]

**Review by guoxbin — APPROVED** [review]

**Pull request merged**